File: typescript/src/patterns/structural/composite/Node.ts
export abstract class Node {
  protected children: Node[] = [];

  constructor(protected readonly name: string) {}

  addChild(node: Node): void {
    this.children.push(node);
  }

  abstract printInfo(): void;

  setChildren(children: Node[]): void {
    this.children = children;
  }
}

File: csharp/src/Training.Patterns/Structural/Composite/Node.cs
namespace Training.Patterns.Structural.Composite;

public abstract class Node
{
    protected readonly string name;
    protected List<Node> children = [];

    public virtual void AddChild(Node node)
    {
        children.Add(node);
    }

    public abstract void PrintInfo();

    protected Node(string name)
    {
        this.name = name;
    }

    public void SetChildren(List<Node> children)
    {
        this.children = children;
    }
}
